src/WCSimIBDGen.cc

```diff
@@ -104,6 +111,37 @@ double WCSimIBDGen::InterpolateSpectrum(std::vector<float> ener_vec, std::vector
     return flux_vec.back();
 }
 
+double WCSimIBDGen::MaxXSecFlux() {
+    // Loops over the energies in the flux model and calculates the maximum value of dSigma/dCosTheta * flux.
+    double xs_flux_max = 0.0;
+
+    for (float ene : energy) {
+        // Calculate the cross section for forward scattered positrons and backward scattered positrons. One of
+        // these values will always represent the maximum cross section across the energy range (forward at high
+        // energies and backward at low energies)
+        double x_sec_forward = dSigmaBydCosTheta(ene, 1.0);
+        double x_sec_backward = dSigmaBydCosTheta(ene, -1.0);
+
+        // Get the flux at ene
+        double flux_at_ene = InterpolateSpectrum(energy, flux, ene);
```

Comment:
Is there a good reason why `InterpolateSpectrum()` takes arguments for `ener_vec` & `flux_vec`?
The only places `InterpolateSpectrum()` is called, member variables `energy` & `flux` are passed.
So why not just use the `energy` and `flux` data members in that method, and remove them as arguments?
If there is a good reason for the two extra arguments, please change it to passing the vectors by (const) reference

Reply:
Thanks for the comments!

I've added the plot of the comparison between this cross section calculation and the SNTools one to the PR description.

There wasn't really a good reason for passing vectors to the interpolation function. I think the original thinking was that the function might also be used to interpolate other vectors but this didn't end up being the case. I've updated this to use the member variables.